### ISSUE TITLE
B903 reword message when a subclass is involved

### ISSUE THREAD

**Issue by ezyang** [PyCQA/flake8-bugbear#63]
The current B903 message reads:

B903 Data class should either be immutable or use __slots__ to save memory. Use collections.namedtuple to generate an immutable class, or enumerate the attributes in a __slot__ declaration in the class to leave attributes mutable.

However, this lint message is not totally correct if the data class in question *inherits* from another class; e.g., Exception; in this case, use of namedtuple is inappropriate (actually, I'm not sure if there's any other way to do it.)

**Comment by tobias-farrenkopf** [comment]
We ran into the very same issue.

**Comment by olliemath** [comment]
Yes, this is idiomatic python, which means we had to disable this check
```python
class MyError(ValueError):
    """MyError is raised when bla bla bla..."""
```

**Comment by olliemath** [comment]
I think that's the most sensible option. It's not perfect, but you'd still catch simple dataclasses and leave more advanced types alone.

Since you can't inspect the MRO in the syntax tree without a lot of effort, it's not practical to do anything more advanced

**Comment by jenstroeger** [comment]
> Want to check for this inheritance and disable the error if that's found? We would need to clearly explain it in the documentation.

Checking inheritance would make sense. In my case I came across a
```python
    class MockStripeObj(dict):
        def __init__(self, param):
            self.id = param
            self.type = param
```
which triggered this opinionated warning and, like above, I’m silencing it as a false positive. If there were no base class(es) (i.e. implicit `object`) then the warning would be valid.

> We would need to clearly explain it in the documentation.

Yup. Probably would make sense to mention something like, “Is this meant to be a dataclass?” or some such?

**Comment by tolomea** [comment]
I'm also hitting this on Exception classes

**Comment by tolomea** [comment]
super in the init shuts it up...
